internal/pkg/cli/deploy/deploy.go

```diff
@@ -394,14 +394,14 @@ func NewWorkerSvcDeployer(in *WorkloadDeployerInput) (*workerSvcDeployer, error)
 
 // UploadArtifactsOutput is the output of UploadArtifacts.
 type UploadArtifactsOutput struct {
-	ImageDigest string
+	ImageDigest *string
```

Comment:
🖐🏼  Question boss! Why pointer?

Reply:
Because since we want to keep the behavior to avoid backward incompatible changes, we need `ImageDigest` to be "" so that it would know this image needs to be build and push (it's just the upload artifact flag is not set), whereas `ImageDigest` to be `nil` means it doesn't need to.

The change of `UploadArtifactsOutput` is for the ease of using (also aligns with our intention that "" is a valid zero value for `UploadArtifactsOutput`).

Reply:
oohh do you mean it's possible that:
1. image has been built and pushed
2. imageDigest is ""

to happen at the same time? I thought `imageDigest` is empty if and only if no image is built

Reply:
if `imageDigest` is empty then we keep using the old URL without pushing the image (example: "1234567890.dkr.ecr.us-west-1.amazonaws.com/demo/frontend:latest") it was actually a bad behavior because it won't work and the url would be different if we are not using the latest tag.

if it is not empty we build and push the image and use the correct URL. 

if it is nil then we do nothing.

Reply:
Comment added so that future maintainer won't remove it accidentally